### ISSUE TITLE
Add proxy for webhook httpclient

### ISSUE THREAD

**Issue by YanBelotskiy** [Xabaril/AspNetCore.Diagnostics.HealthChecks#243]
For use proxy
Possible changes:

Change for HealthChecks.UI/Keys.cs
Add 
internal const string WEBHOOK_HTTP_CLIENT_NAME = "webhook";

Change for HealthChecks.UI/Core/Notifications/WebHookFailureNotifier.cs
Add
private readonly HttpClient _httpClient;

change .ctor like this
public WebHookFailureNotifier(HealthChecksDb db, IOptions<Settings> settings,
            IHttpClientFactory httpClientFactory, ILogger<WebHookFailureNotifier> logger)
        {
            _db = db ?? throw new ArgumentNullException(nameof(db));
            _settings = settings.Value ?? new Settings();
            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
            _httpClient = httpClientFactory.CreateClient(Keys.WEBHOOK_HTTP_CLIENT_NAME);
        }

change 
private async Task SendRequest(string uri, string name, string payloadContent)
        {
            if (uri == null || !Uri.TryCreate(uri, UriKind.Absolute, out Uri webHookUri))
            {
                _logger.LogWarning($"The web hook notification uri is not established or is not an absolute Uri ({name}). Set the webhook uri value on BeatPulse settings.");

                return;
            }
            try
            {
                var payload = new StringContent(payloadContent, Encoding.UTF8, Keys.DEFAULT_RESPONSE_CONTENT_TYPE);
                var response = await _httpClient.PostAsync(webHookUri, payload);
                if (!response.IsSuccessStatusCode)
                {
                    _logger.LogError($"The webhook notification has not executed successfully for {name} webhook. The error code is {response.StatusCode}.");
                }
            }
            catch (Exception exception)
            {
                _logger.LogError($"The failure notification for {name} has not executed successfully.", exception);
            }
        }

**Comment by CarlosLanderas** [comment]
You can already use a Proxy in HealthChecks.UI, you just need to pass a custom HttpHandler.
Do you need to have the proxy only configured for webhooks and not for collecting health checks in remote endpoints?

Sample:
```csharp
 var handler  = new HttpClientHandler()
 {
    Proxy = 
    PreAuthenticate = ,
    UseDefaultCredentials = 
 };

 services
     .AddHealthChecksUI(setupSettings: setup => {
           setup.UseHttpMessageHandler(handler);
     });

```
PR: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/211

**Comment by YanBelotskiy** [comment]
yes, I need to have the proxy only configured for webhooks uris and not for collecting health checks in remote endpoints

**Comment by mnadeau2** [comment]
same issue

**Comment by CarlosLanderas** [comment]
I'm gonna fix today #256 and use the time to have separate http handlers for the api and the webhooks client.

**Comment by CarlosLanderas** [comment]
UI Version 2.2.33 is being published. You have a usage sample here:

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/test/FunctionalTests/HealthChecks.UI/Configuration/UIConfigurationTests.cs#L171-L190
